packages/react-storage/src/components/StorageBrowser/context/controls/Paginate/Paginate.tsx

```diff
@@ -0,0 +1,83 @@
+import React from 'react';
+
+const INITIAL_STATE: PaginateState = {
+  current: 1,
+  hasNext: false,
+  hasPrevious: false,
+  getPageData: () => undefined,
+  isLoadingNextPage: false,
+};
+
+type Permission = 'READ' | 'READWRITE' | 'WRITE';
+type LocationType = 'OBJECT' | 'PREFIX' | 'BUCKET';
+
+// avoid naming clash with Location (URL) API (https://developer.mozilla.org/en-US/docs/Web/API/Location)
+interface LocationData {
+  name: string;
+  permission: Permission;
+  type: LocationType;
+}
+
+interface FolderData {
+  key: string;
+  type: 'FOLDER';
+}
+
+interface FileData {
+  key: string;
+  lastModified: Date;
+  size: number;
+  type: 'FILE';
+}
+
+type LocationItem = FileData | FolderData;
+
+interface ListData {
+  LOCATIONS: LocationData;
+  LOCATION_ITEMS: LocationItem;
+}
```

Comment:
Should we move these interfaces into a shared place? I'm sure this might already be on your radar, but I thought it would be helpful to track here